examples/temperature/temperature-service-console-client/temperatureserviceinterface.cpp

```diff
@@ -22,42 +22,50 @@
 
 #include "temperatureserviceinterface.h"
 
-TemperatureServiceInterface::TemperatureServiceInterface(DBus::Connection &connection,
-                                                         std::string logfilepath) :
-    m_logfilepath(logfilepath),
-    DBus::ObjectProxy(connection,
-                      "/com/pelagicore/TemperatureService",
-                      "com.pelagicore.TemperatureService")
-{
+void TemperatureServiceInterface::proxy_callback(const Glib::RefPtr<Gio::AsyncResult> &result) {
+    m_proxy = com::pelagicore::TemperatureService::createForBusFinish(result);
 }
 
-std::string TemperatureServiceInterface::echo(const std::string &str)
+TemperatureServiceInterface::TemperatureServiceInterface()
 {
-    std::cout << str << std::endl;
-    return str;
+    com::pelagicore::TemperatureService::createForBus(Gio::DBus::BUS_TYPE_SESSION,
+                                                      Gio::DBus::PROXY_FLAGS_NONE,
+                                                      "com.pelagicore.TemperatureService",
+                                                      "/com/pelagicore/TemperatureService",
+                                                      sigc::mem_fun(*this, &TemperatureServiceInterface::proxy_callback));
+}
+
+void TemperatureServiceInterface::on_echo_finished(const Glib::RefPtr<Gio::AsyncResult> &result) {
+    std::string  echoReturn;;
+    m_proxy->Echo_finish(echoReturn, result);
+    std::cout << "Echo : " << echoReturn;
 }
 
-double TemperatureServiceInterface::getTemperature()
+void TemperatureServiceInterface::echo(const std::string &str)
 {
-    return GetTemperature();
+    m_proxy->Echo(str, sigc::mem_fun(*this ,&TemperatureServiceInterface::on_echo_finished));
 }
 
-void TemperatureServiceInterface::setTemperature(const double &temperature)
+void TemperatureServiceInterface::on_gettemp_finished(const Glib::RefPtr<Gio::AsyncResult> &result) {
+    double temperature;
+    m_proxy->GetTemperature_finish(temperature, result);
+    std::cout << "Temperature : " << temperature << std::endl;
+}
+
+void TemperatureServiceInterface::getTemperature()
 {
-    SetTemperature(temperature);
+    m_proxy->GetTemperature(sigc::mem_fun(*this,
+                                          &TemperatureServiceInterface::on_gettemp_finished));
```

Comment:
Would be nice to extract the `mem_fun` stuff to `auto callback = sigc::mem_fun...`